src/libexpr/primops/fetchTree.cc

```diff
@@ -321,7 +321,8 @@ static RegisterPrimOp primop_fetchTree({
 
         - `shallow` (Bool, optional)
 
-          Make a shallow clone when fetching the Git tree.
+          Allow fetching from a shallow remote repository.
+          This still performs a full clone of what is available on the remote.
```

Comment:
Actually this behavior changed recently with the libgit2 refactor. It now actually does a shallow fetch when shallow is set `true`.
That implies, whenever shallow=true:
- the revCount will always be `1`

Basically, there two lines can be reverted and instead the other instance of on fetchGit needs to be corrected

Reply:
I think `revCount` should be unset for shallow fetches.